### ISSUE TITLE
npm -g install stf get an error

### ISSUE THREAD

**Issue by gam2046** [openstf/stf#1111]
There is a similar problem here: #950  . But nodejs, which I have already used 8.x, still have problems.

```
npm WARN deprecated hoek@2.16.3: This version has been deprecated in accordance with the hapi support policy (hapi.im/support). Please upgrade to the latest version to get the best features, bug fixes, and security patches. If you are unable to upgrade at this time, paid support is available for older versions (hapi.im/commercial).
npm WARN deprecated ejs@0.8.8: Critical security bugs fixed in 2.5.5
/usr/local/bin/stf -> /usr/local/lib/node_modules/stf/bin/stf

> dtrace-provider@0.8.8 install /usr/local/lib/node_modules/stf/node_modules/dtrace-provider
> node-gyp rebuild || node suppress-error.js

gyp WARN EACCES user "undefined" does not have permission to access the dev dir "/root/.node-gyp/8.16.2"
gyp WARN EACCES attempting to reinstall using temporary dev dir "/usr/local/lib/node_modules/stf/node_modules/dtrace-provider/.node-gyp"
gyp WARN install got an error, rolling back install
gyp WARN install got an error, rolling back install
gyp ERR! configure error
gyp ERR! stack Error: EACCES: permission denied, mkdir '/usr/local/lib/node_modules/stf/node_modules/dtrace-provider/.node-gyp'
gyp ERR! System Linux 4.9.0-8-amd64
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /usr/local/lib/node_modules/stf/node_modules/dtrace-provider
gyp ERR! node -v v8.16.2
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok

> bufferutil@1.3.0 install /usr/local/lib/node_modules/stf/node_modules/bufferutil
> node-gyp rebuild

gyp WARN EACCES user "undefined" does not have permission to access the dev dir "/root/.node-gyp/8.16.2"
gyp WARN EACCES attempting to reinstall using temporary dev dir "/usr/local/lib/node_modules/stf/node_modules/bufferutil/.node-gyp"
gyp WARN install got an error, rolling back install
gyp WARN install got an error, rolling back install
gyp ERR! configure error
gyp ERR! stack Error: EACCES: permission denied, mkdir '/usr/local/lib/node_modules/stf/node_modules/bufferutil/.node-gyp'
gyp ERR! System Linux 4.9.0-8-amd64
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /usr/local/lib/node_modules/stf/node_modules/bufferutil
gyp ERR! node -v v8.16.2
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! bufferutil@1.3.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the bufferutil@1.3.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

**Comment by Coder2cdb** [comment]
I have the same trouble,and I do not know how to solve this problem!

**Comment by gam2046** [comment]
> I have the same trouble,and I do not know how to solve this problem!

Maybe it will help you

Here is my Dockerfile:
https://github.com/gam2046/dockers/blob/master/stf/Dockerfile

Of course, you can also directly pull the image
`docker pull gam2046/stf`

**Comment by Dezzpil** [comment]
**TL;DR** npm install --unsafe-perm
